test/integration/cds_integration_test.cc

```diff
@@ -164,6 +102,25 @@ class CdsIntegrationTest : public Grpc::GrpcClientIntegrationParamTest, public H
     test_server_->waitUntilListenersReady();
     registerTestServerPorts({"http"});
   }
+
+  void acceptXdsConnection() {
+    AssertionResult result = // xds_connection_ is filled with the new FakeHttpConnection.
+        fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, xds_connection_);
+    RELEASE_ASSERT(result, result.message());
+    result = xds_connection_->waitForNewStream(*dispatcher_, xds_stream_);
+    RELEASE_ASSERT(result, result.message());
+    xds_stream_->startGrpcStream();
+    fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+  }
+
+  virtual void giveInitialCluster() {
```

Comment:
Can you add some comments here? What does this do?

Reply:
Done.